browserstack-examples-appium-nunit/Browserstack/App/Tests/ReportGeneration.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+
+using NUnit.Framework;
+using Newtonsoft.Json.Linq;
+using System.Threading.Tasks;
+using RestSharp;
+using RestSharp.Authenticators;
+using BrowserStack.App.Pages;
+using System.IO;
+
+namespace browserstack_examples_appium_nunit.Browserstack.App.Tests
+{
+    public class ReportGeneration
+    {
+        [Test]
+        public async Task ReportGenerator()
+        {
+            var BuildId = "";
+            var Limit = 100;
+            var Offset = 0;
+            var userName = Environment.GetEnvironmentVariable("BROWSERSTACK_USERNAME") ?? "";
+            var accessKey = Environment.GetEnvironmentVariable("BROWSERSTACK_ACCESS_KEY") ?? "";
+            var buildName = Environment.GetEnvironmentVariable("BROWSERSTACK_BUILD_NAME") ?? "";
+            //var buildName = "Sales Demo";
+
+
+            var fileLocation = "../../../../Reports/output.html";
+            string[] htmlStart =
+                {
+        "<!DOCTYPE html><html><head><style>table {  font-family: arial, sans-serif; border-collapse: collapse; width: 100%;}td, th{ border: 1px solid #dddddd;  text-align: left;  padding: 8px;}</style></head><body><table>",
+        "<tr><th>Build Name</th><th>Project Name</th><th>Session ID</th><th>Session Name</th><th>OS Name</th><th>OS Version</th><th>Device</th><th>Status</tr>"
+
+        };
+
+            string[] htmlEnd = { "</table></body></html>" };
+
+            await System.IO.File.WriteAllLinesAsync(fileLocation, htmlStart);
+
+            var client = new RestClient("https://api.browserstack.com");
+            client.Authenticator = new HttpBasicAuthenticator(userName, accessKey);
+
+            var buildApiRequest = new RestRequest("https://api.browserstack.com/app-automate/builds.json?limit=40");
```

Comment:
Refer this part `https://api.browserstack.com` from the variable you have defined earlier

Reply:
Done